app/views/layouts/settings.html.erb

```diff
@@ -16,11 +16,29 @@
     Settings
   </header>
   <section id="menu" data-controller="radio-behavior" data-radio-behavior-selected-class="relationship">
-    <% @settings_menu.each do |controller, settings_items| %>
-      <div class="mb-5">
-        <%= render partial: "layouts/settings_item", collection: settings_items, locals: { controller: controller } %>
-      </div>
-    <% end %>
```

Comment:
2) You can go back to this and just change it to `each_with_index do | (controller, settings_item), i |`. Then within the div just add an extra class inline `<div class="mb-5 <%= (i > 5) && @hide_assistant_overflow && 'hidden' %>" data-transition-target="transitionable">`
3) On the `<section ..>` you can add the transitions controller and wire it up just like you have so that it simultaneously hides the "show remaining" button while unhiding the extra assistants

This approach would save eliminate multiple branches of logic.

I hope this doesn't come as being too nit picky! :) I don't actually feel that strongly about it so you could also keep it as is. I just thought of this alternate approach as I was reading through things.

Reply:
Not nit-picky. Happy to improve the implementation when I return to this (Sunday?)

Reply:
Is there a mixup? The `i>5` would not apply because there are only 2 controllers (assistant/people). For each controller, there is one `div` with `class=mb-5`.  So I applied the _idea_ of your suggestion. Also to the "user-facing side" (messages), where it seems it is not needed to expand the list when a otherwise hidden assistant is active. A bit tricky in that way.